### DISCUSSION TITLE
Provide jet-smearing (including random numbers) within correctionlib

### DISCUSSION THREAD

**Discussion by kirschen** [cms-nanoAOD/correctionlib#123] [Ideas]
Hi @nsmith- , all,

@lgray brought up the idea of going a step further in what is provided via correctionlib. One classical example of where one would profit from harmonizing is jet energy smearing, see e.g. https://twiki.cern.ch/twiki/bin/viewauth/CMS/JetResolution#Smearing_procedures
Depending on whether or not a genjet is matched (well enough) to a recojet, either the difference between genjet pt and recojet pt is changed or some stochastic smearing is applied. Ideally, the result would be reproducible, e.g. deciding on a standard way to calculate a random number seed (e.g. requiring event/lumisection as input).

Input would be:
recopt, recoeta, genpt, (info for random number seed)
Output: 
smeared recopt

Would this still be doable and within scope of correctionlib @nsmith- ?

Cheers,
Henning

**Comment by nsmith-** [comment]
I think since it is still just a scalar function then that meets the scope of the project. We can have either a special node or simply a special function available to formula nodes, which can generate deterministic pseudorandom numbers.
This was discussed a bit also for coffea JEC a while back in https://github.com/CoffeaTeam/coffea/issues/454 and there I proposed using some combination of event and jet properties as an entropy source for use in a [integer hashing function](https://stackoverflow.com/questions/664014/what-integer-hash-function-are-good-that-accepts-an-integer-hash-key).

**Comment by nsmith-** [comment]
This discussion was converted to issue #130 and implemented with PR #140
